modules/monitors/main.tf

```
@@ -6,7 +6,7 @@ locals {
 
 # https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor
 resource "datadog_monitor" "default" {
-  for_each = local.enabled ? { for k, v in var.datadog_monitors : k => v if lookup(v, "enabled", true) } : {}
+  for_each = local.enabled ? { for k, v in var.datadog_monitors : k => v if lookup(v, "enabled", true) } : null
```

I believe this isn't valid, it will generate the following error:

```
The given "for_each" argument value is unsuitable: the given "for_each"
│ argument value is null. A map, or set of strings is allowed.
```

---

Somehow missed @Gowiem 's [comment](https://github.com/cloudposse/terraform-datadog-platform/pull/66#issuecomment-1192940666) - he's already flagged it.

I have seen the error you originally reported when the dictionary keys don't match for each monitor definition. Could that describe your issue?

---

Hi @joe-niland 
we got this error
```
The true and false result expressions must have consistent types. The 'true'
│ value includes object attribute "test", which is
│ absent in the 'false' value.
```
the problem is that {"test": ...} and {} are different types in TF. But with setting {} to null, it works fine for us (TF version is 1.1.7)

---

@olhabilan There must be something different between TF `1.2.5` and `1.1.7`... 

Anyway, can you try the following update?

```suggestion
  for_each = local.enabled ? tomap({ for k, v in var.datadog_monitors : k => v if lookup(v, "enabled", true) }) : {}
```

Let's see if that works for you + works for these tests. Terraform is treating the map that we're building via the loop as an `object` type and we just need it to deal with it as a `map` type, so that the truthy and falsey values match types. I'm hoping `tomap` will do the correct type conversion for it and make it shut up 😆

---

thanks for the idea
but I got another error 
`var.datadog_monitors is object with 17 attributes
│ 
│ Invalid value for "v" parameter: cannot convert object to map of any single
│ type.`

I think maybe we pass incorrect data to datadog_monitor variable 🤔